pkgs/development/compilers/swi-prolog/default.nix

```diff
@@ -4,7 +4,32 @@
 , cmake, libyaml, Security
 , libjpeg, libX11, libXext, libXft, libXinerama
 , extraLibraries ? [ jdk unixODBC libXpm libSM libXt freetype fontconfig ]
-, extraPacks     ? []
+# Packs must be installed from a local directory during the build, with dependencies
```

Comment:
Not sure this is the right approach.

Reply:
Yeah it's pretty bad.

The main solution I've seen other languages use is to have the language's own package manager resolve dependency versions and write a lockfile that a nix-based tool can use to define the environment.

However in this PR I just wanted to set up a baseline that works, even if manual and smelly.

Reply:
@SuperSandro2000 is there a preferred method I can aim for? I'm open to ideas.

Reply:
> However in this PR I just wanted to set up a baseline that works, even if manual and smelly.

I would prefer if it would be done correct the first time but it was already like this and you just added some very needed documentation so I think this should be fine.